### PR TITLE
vtk: Added mpi variant

### PR DESCRIPTION
**Pull request by bilke**

### CODE REVIEW

File: var/spack/repos/builtin/packages/vtk/package.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ class Vtk(CMakePackage):
     variant('osmesa', default=False, description='Enable OSMesa support')
     variant('python', default=False, description='Enable Python support')
     variant('qt', default=False, description='Build with support for Qt')
+    variant('mpi', default=False, description='Enable MPI support')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/vtk/package.py]: `mpi` variants should be default-`True`: https://spack.readthedocs.io/en/latest/packaging_guide.html#variant-names

File: var/spack/repos/builtin/packages/vtk/package.py [hunk 2]
```diff
@@ -73,6 +74,8 @@ class Vtk(CMakePackage):
 
     # VTK will need Qt5OpenGL, and qt needs '-opengl' for that
     depends_on('qt+opengl', when='+qt')
+    
+    depends_on('mpi', when='+mpi')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/vtk/package.py]: do you know by coincidence which standard? MPI 2.3+? (`mpi@2.3:`)

File: var/spack/repos/builtin/packages/vtk/package.py [hunk 3]
```diff
@@ -73,6 +74,8 @@ class Vtk(CMakePackage):
 
     # VTK will need Qt5OpenGL, and qt needs '-opengl' for that
     depends_on('qt+opengl', when='+qt')
+
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/vtk/package.py]: flake8 error on this line (trailing whitespaces)
https://travis-ci.org/spack/spack/jobs/440071587

File: var/spack/repos/builtin/packages/vtk/package.py [hunk 4]
```diff
@@ -167,6 +170,12 @@ def cmake_args(self):
                     '-DModule_vtkGUISupportQtOpenGL:BOOL=ON',
                 ])
 
+        if '+mpi' in spec:
+            cmake_args.extend([
+                '-DVTK_Group_MPI:BOOL=ON',
+                '-DVTK_USE_SYSTEM_DIY2=OFF'
+            ])
+
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/vtk/package.py]: flake8 error on this line (trailing whitespaces)
https://travis-ci.org/spack/spack/jobs/440071587

**Comment by ax3l** [comment]
potential collision with changes in #9510

File: var/spack/repos/builtin/packages/vtk/package.py (hunk 2 above)
Comment (bilke) [reply var/spack/repos/builtin/packages/vtk/package.py]: No sorry I have not found any info on this.

**Comment by bilke** [comment]
@ax3l I addressed your comments.

**Comment by tgamblin** [comment]
@bilke: can you please see #9137 and add your consent?  Let me know if you have questions.

**Pull request merged**